Services/Database/classes/Atom/class.ilAtomQueryBase.php

```diff
@@ -91,21 +92,22 @@ public function getRisks() {
 
 
 	/**
-	 * Add table-names which are influenced by your queries, MyISAm has to lock those tables. Lock
+	 * Add table-names which are influenced by your queries, MyISAm has to lock those tables.
+	 * You get an ilTableLockInterface with further possibilities, e.g.:
 	 *
-	 * @param string $table_name
-	 * @param bool $lock_sequence_too
-	 * @throws \ilAtomQueryException
+	 * $ilAtomQuery->addTableLock('my_table')->lockSequence(true)->aliasName('my_alias');
+	 *
+	 * the lock-level is determined by ilAtomQuery
+	 *
+	 * @param $table_name
+	 * @return \ilTableLockInterface
 	 */
-	public function lockTable($table_name, $lock_sequence_too = false) {
-		if (!$table_name || !$this->ilDBInstance->tableExists($table_name)) {
-			throw new ilAtomQueryException('', ilAtomQueryException::DB_ATOM_LOCK_TABLE_NONEXISTING);
-		}
-		$lock_level = $this->getDeterminedLockLevel();
-		if (!in_array($lock_level, array( ilAtomQuery::LOCK_READ, ilAtomQuery::LOCK_WRITE ))) {
-			throw new ilAtomQueryException('', ilAtomQueryException::DB_ATOM_LOCK_WRONG_LEVEL);
-		}
-		$this->tables[] = array( $table_name, $lock_level, $lock_sequence_too );
+	public function addTableLock($table_name) {
```

Comment:
I don't see the point in renaming the method, was there any specific reason? It's mostly for curiosity, although i'm a little biased to the old name.